### PR TITLE
Structure text

### PR DESCRIPTION
**Pull request by tripplybrand**
Add variant prop to Text.js so the texts laid out in Figma can be selected: h1-h6, bodyLarge, bodyBase, bodySmall, and label

### CODE REVIEW

File: src/Button/Button.js [hunk 1]
```diff
@@ -22,15 +22,15 @@ function Button(props, ref) {
...
-        `${background} ${borderWidth} ${colors.border}`
+        `${background} ${borderWidth} ${colors.border}`,
+        className
       )}
       {...mergeProps(buttonProps, hoverProps, focusProps)}
       disabled={disabled}
     >
       {typeof children === 'string' ? (
-        <Text className="text-lg font-normal">{children}</Text>
+        <Text variant="h4">{children}</Text>
```
Comment (brookslybrand) [inline src/Button/Button.js:33]: I'm not 100% sure if it's correct to use semantic html such as headers inside of buttons

File: src/Text/Text.js [hunk 2]
```diff
@@ -1,19 +1,68 @@
...
-      className={classNames('font-display text-base', className)}
+      className={classNames('text-gray-pink-400', className)}
     >
       {children}
     </Component>
   )
 }
 
+const getHTMLElement = variant => {
+  if (
```
Comment (brookslybrand) [inline src/Text/Text.js:55]: Whenever I have a situation like this, I like to write it as `['h1','h2','h3','h4','h5','h6',].includes(variant)`. You definitely don't have too, I just find it a little easier.

File: src/Text/Text.js [hunk 3]
```diff
@@ -1,19 +1,68 @@
...
 import classNames from 'classnames'
 
-// this component is not fully fleshed out, it's just here to add some typography
-function Text({ className, as, children }, ref) {
-  const Component = as ?? 'span'
+// this is the variant approach to text
+function Text({ as, variant, children }, ref) {
+  const Component = as ?? getHTMLElement(variant)
+
+  var className
```
Comment (brookslybrand) [inline src/Text/Text.js:8]: Nit: prefer let to var. We don't have a lint rule about this, so maybe we should, though I don't want to be overly pedantic if someone has a good reason to use var.

File: src/Text/Text.js [hunk 4]
```diff
@@ -1,19 +1,68 @@
...
+  switch (variant) {
```
Comment (brookslybrand) [inline src/Text/Text.js:9]: Nit: large code blocks like this should be abstracted out of the component

File: src/Button/Button.js (hunk 1 above)
Comment (tripplybrand) [reply src/Button/Button.js:33]: Ah, well that's good to know. If that's true I think I'll need to make bodyLarge Poppins then.

File: src/Text/Text.js (hunk 2 above)
Comment (tripplybrand) [reply src/Text/Text.js:55]: Cool. I just copied from what we are doing in Calumet, so clearly you didn't write that code either, haha

Comment (brookslybrand) [reply src/Text/Text.js:55]: Haha ah, I was a sillier person back then, who liked to make things more difficult for myself

**Pull request merged**